### PR TITLE
Discard unused locals cleanup

### PR DESCRIPTION
**Pull request by vcsjones**
Contributes to #39962.

/cc @bartonjs

### CODE REVIEW

File: src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/SignedXml.cs [hunk 1]
```diff
@@ -409,7 +409,7 @@ public void ComputeSignature()
             HashAlgorithm hashAlg = signatureDescription.CreateDigest();
             if (hashAlg == null)
                 throw new CryptographicException(SR.Cryptography_Xml_CreateHashAlgorithmFailed);
-            byte[] hashvalue = GetC14NDigest(hashAlg);
+            _ = GetC14NDigest(hashAlg);
```
Comment (vcsjones) [inline src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/SignedXml.cs]: The gist of this one is that `hashAlg` carries state and `GetC14NDigest` is being used to update the hash, as well as cache it. The return value isn't used but the call needs to be made still to update the hash, which is used a little below here in the signature formatter.

The whole `GetC14NDigest` could probably be refactored to be more clear, but there are a lot of dragons in the file and I'd likely just break something.

File: src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/StorePal.Import.cs [hunk 2]
```diff
@@ -164,7 +164,7 @@ public static IStorePal FromSystemStore(string storeName, StoreLocation storeLoc
             //
             // For compat with desktop, ignoring any failures from this call. (It is pretty unlikely to fail, in any case.)
             //
-            bool ignore = Interop.crypt32.CertControlStore(certStore, CertControlStoreFlags.None, CertControlStoreType.CERT_STORE_CTRL_AUTO_RESYNC, IntPtr.Zero);
+            _ = Interop.crypt32.CertControlStore(certStore, CertControlStoreFlags.None, CertControlStoreType.CERT_STORE_CTRL_AUTO_RESYNC, IntPtr.Zero);
```
Comment (vcsjones) [inline src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/StorePal.Import.cs]: Change justified by comment above.

File: src/Common/src/Interop/Windows/Advapi32/SafeProvHandle.cs [hunk 3]
```diff
@@ -110,7 +110,7 @@ protected override bool ReleaseHandle()
 
                 uint flags = (_flags & (uint)Interop.Advapi32.CryptAcquireContextFlags.CRYPT_MACHINE_KEYSET) | (uint)Interop.Advapi32.CryptAcquireContextFlags.CRYPT_DELETEKEYSET;
                 SafeProvHandle hIgnoredProv;
-                bool ignoredSuccess = Interop.Advapi32.CryptAcquireContext(out hIgnoredProv, _containerName, _providerName, _type, flags);
+                _ = Interop.Advapi32.CryptAcquireContext(out hIgnoredProv, _containerName, _providerName, _type, flags);
```
Comment (vcsjones) [inline src/Common/src/Interop/Windows/Advapi32/SafeProvHandle.cs]: Change justified by comment below.

**Review by bartonjs — APPROVED** [review]

**Comment by stephentoub** [comment]
Why use discards for these at all? If we're not going to explicitly call out via the "success" naming that the results represent success, what benefit is the "_ = " providing? Why not just remove those entirely? We do not have analyzers enabled to remove consumption of every return value that goes unused... if we did I expect we'd have thousands of errors across corefx.

**Comment by vcsjones** [comment]
@stephentoub it's mostly a preference of my own coding style. I'm happy enough to remove them if you think they are not helpful. My general attitude for discards, especially error conditions in cryptographic areas, is to be explicit as possible.

The `ignore` and `ignoreSuccess` locals weren't required in the first place, I assumed they were there and named such to provide a similar type of explicitness that could not be done before the availability of discards in C#.

**Comment by bartonjs** [comment]
The two "ignored" ones feel like either the name or the discard are useful (shows intent).

The one in SignedXml would probably be better as a comment saying that the side effect is what's needed (effectively the comment you left here), and then no discard or variable. For some reason I thought that one was in a test.

**Comment by stephentoub** [comment]
> the discard are useful (shows intent)

Does that mean every place where we simply ignore a return value is an accident / lacks intent?

**Comment by bartonjs** [comment]
I don't think it's useful to do it to all ignored method returns, no.  But the two that used to have variables ignored are P/Invokes, and ignoring return values in Win32 is a pretty good source of bugs.  So I think the discards there indicate for posterity (along with the comment) that effectively says "yes, I know I should care about the return value from Win32 API, but in this case it's right to not care".

The one from SignedXml doesn't meet my personal neural net flow for "say why I'm ignoring the return value"; but it does meet my "this code looks like it's not needed, so a comment should indicate that we're after a side-effect of this Get method".

**Comment by stephentoub** [comment]
> the two that used to have variables named "ignored" are P/Invokes, and ignoring return values in Win32 is a pretty good source of bugs. So I think the discards there indicate for posterity (along with the comment) that effectively says "yes, I know I should care about the return value from Win32 API, but in this case it's right to not care".

Ok. Just seems strange to me to use a discard for that when there's already a long comment for each explaining that and why the result is being ignored. I don't see what the discard adds here.

Up to you, though.

Thanks.

**Review by stephentoub — APPROVED** [review]

**Pull request merged**